### ISSUE TITLE
"Ask confirmation before downloading folders larger than X MB" does not work when creating a folder sync connection

### ISSUE THREAD

**Issue by SamuAlfageme** [owncloud/client#5421]
### Expected behavior
After setting a download size limit for folders in the sync client settings:

![larger_than](https://cloud.githubusercontent.com/assets/2644445/21496139/7e59b196-cc1c-11e6-9ffb-62f0cc4a526f.png)

Those remote folders larger than the specified size should be initially unchecked on the wizards.

### Actual behavior
No manual confirmation is asked, everything is selected in the `Choose what to sync` (in both new account and new folder sync connection wizards) and syncs **by default**:

![screenshot_2016-12-27_10_01_36](https://cloud.githubusercontent.com/assets/2644445/21496005/8ccefbec-cc1b-11e6-9919-8497c83d8a9a.png)

However, when the `.csync_journal.db` (or `._sync_[...].db` after #5045) is removed, the expected behavior is triggered:

* The notification is finally displayed:

![screenshot 2016-12-27 09 48 45](https://cloud.githubusercontent.com/assets/2644445/21495744/b739b6a8-cc19-11e6-9fac-2166af283772.png)

* The 3 options to handle these folders are shown: 

![screenshot 2016-12-27 09 52 14](https://cloud.githubusercontent.com/assets/2644445/21495908/c7e88744-cc1a-11e6-8ddf-e82b283cb33f.png)

### Steps to reproduce
1. Configure the sync client to ask confirmation before downloading folders larger than 20MB
2. Create a folder larger than 20MB in the web UI
3. Complete the add account/create new folder sync connection wizard selecting to synchronize a parent the folder just created.

### Client configuration
* Client version: 2.2.4
* Operating system: OS X 10.11.6

This was also reported in https://github.com/owncloud/client/pull/5340#issuecomment-266415969 as this is the workaround used there and causing it to fail on first runs.

**Comment by SamuAlfageme** [comment]
I think this is somehow conflicting with the `Theme::wizardSelectiveSyncDefaultNothing()` concept.

Also related with that: #4581

**Comment by ckamm** [comment]
@SamuAlfageme I don't think this is a bug. Currently the user gets to explicitly choose which folders to sync when the folder sync connection is made. So after the user approves the dialog where all the folders are listed with their sizes, the client doesn't believe that big folders need an additional level of confirmation.

I'm unsure how to resolve this: We could disable these bigger folders by default, but this might be surprising to users. Maybe then the "Choose What To Sync" dialog itself needs a note that "Some folders like A, B, C, ... will not be synced by default because their size exceeds the configured threshold"?

**Comment by phil-davis** [comment]
I guess the client knows the size of each folder at that point when it presents the "Choose What To Sync" dialog. If so, then I would expect it to deselect each folder over the limit, using the same logic as if all those folders had just appeared on the server for new download.

If there is a download confirmation threshold set, and there was at least 1 folder exceeding the threshold, then text like the following could be displayed:
"Folders whose size exceeds the download confirmation threshold (nnnMB) have been deselected. You may select them if you wish to download them."

**Comment by ckamm** [comment]
@phil-davis Agreed, that's what I started on :)

**Comment by ckamm** [comment]
Pull request: https://github.com/owncloud/client/pull/5426

**Comment by guruz** [comment]
(ReadyToTest in PR not master)

**Comment by SamuAlfageme** [comment]
Summing up with this:

* **Now the expected behavior (those folders larger than the threshold are unchecked) when selecting what to sync works** 🎉
* The option "`Ask confirmation before downloading folders larger than 500 MB`" is selected by default in a clean install of the client
* "Sync everything" (without considering those folders that exceed the threshold) is the default behavior in the new account wizard (if `Theme::wizardSelectiveSyncDefaultNothing()` is true), "Choose what to sync" would be more close to what happens in the new folder sync connection wizard
* For the new folders created via web UI exceeding the set size (created one after another, not in bulk) in some weird cases, they are first created empty and then their content is downloaded. I'll see how often this happens and open a new issue if needed. 

Great work!
